File: 06Sort/6-03 삽입정렬.py
#6-3 삽입정렬
arr=[7,5,9,0,3,1,6,2,4,8]
for i in range(1,len(arr)): #첫 번째보다 한칸 앞으로 반복문 설정
    for j in range(i,0,-1): # 그 i부터 반대로 반복문 설정
        if arr[j-1]>arr[j]: #앞에 있는 넘이 더 크다면
            arr[j-1],arr[j]=arr[j],arr[j-1] #스와프
print(arr) #오름차순

arr=[7,5,9,0,3,1,6,2,4,8]
for i in range(1,len(arr)):
    for j in range(i,0,-1):
        if arr[j-1]<arr[j]: #앞에 있는 넘이 작으면
            arr[j-1],arr[j]=arr[j],arr[j-1]
print(arr) #내림차순

File: 06Sort/6-03 삽입정렬.cpp
#include <iostream>
using namespace std;

int n = 10;
int arr[10]= { 7, 5, 9, 0, 3, 1, 6, 2, 4, 8 };

int main(void) {
	for (int i = 1; i < n; i++) {
		//i부터 1까지 감소
		for (int j = i; j > 0; j--) {
			if (arr[j] < arr[j - 1]) {
				swap(arr[j], arr[j - 1]);
			}
			//자기보다 작은 값을 만나면 그 위치에서 멈춤
			else break;
		}
	}
	for (int i = 0; i < n; i++) {
		cout << arr[i] << ' ';
	}
	cout << '\n';
}